src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1316,6 +1318,111 @@ testUnsafeSendChan TestTransport{..} = do
 
   takeMVar clientDone
 
+testRegistryMonitoring :: TestTransport -> Assertion
+testRegistryMonitoring TestTransport{..} = do
+  localNode <- newLocalNode testTransport initRemoteTable
+  remoteNode <- newLocalNode testTransport initRemoteTable
+  return ()
+
+  -- Local process. Test if local process will be removed from
+  -- registry when it dies.
+  box <- newEmptyMVar
+  runProcess localNode $ do
+    pid <- spawnLocal $ do
+                expect
+    register "test" pid
+    tpid <- whereis "test"
+    if tpid == Just pid
+      then do _ <- monitor pid
+              send pid ()
+              ProcessMonitorNotification{} <- expect
+              tpid1 <- whereis "test"
+              liftIO $ putMVar box (Nothing == tpid1)
+      else liftIO $ putMVar box False
+  True <- takeMVar box     
+  return ()
+
+  -- Remote process. Test if remote process entry is removed
+  -- from registry when process dies.
+  remote1 <- testRemote remoteNode 
+  runProcess localNode $
+    let waitpoll = do
```

Comment:
I think the property you want to check is that the process gets removed from the registry, whether you monitor it or not. The test for the local test looks good to me. Monitors are independent of each other. So setting a monitor for the purposes of the test shouldn't interfere or otherwise paper over any missing monitors inside the code under test if any.